northstar/src/runtime/process/init.rs

```diff
@@ -71,60 +198,11 @@ impl Init {
         self.drop_privileges();
 
         // Close and dup fds
-        self.file_descriptors();
-
-        // Clone
-        match unsafe { unistd::fork() } {
-            Ok(result) => match result {
-                unistd::ForkResult::Parent { child } => {
-                    // Drop checkpoint. The fds are cloned into the child and are closed upon execve.
-                    drop(condition_notify);
-
-                    // Wait for the child to exit
-                    loop {
-                        match waitpid(Some(child), None) {
-                            Ok(WaitStatus::Exited(_pid, status)) => {
-                                let exit_status = ExitStatus::Exit(status);
-                                exit_status_channel
-                                    .send(&exit_status)
-                                    .expect("Failed to send exit status");
-                                exit(0);
-                            }
-                            Ok(WaitStatus::Signaled(_pid, status, _)) => {
-                                let exit_status = ExitStatus::Signalled(status as u8);
-                                exit_status_channel
-                                    .send(&exit_status)
-                                    .expect("Failed to send exit status");
-                                exit(0);
-                            }
-                            Ok(WaitStatus::Continued(_)) | Ok(WaitStatus::Stopped(_, _)) => {
-                                continue
-                            }
-                            Err(nix::Error::EINTR) => continue,
-                            e => panic!("Failed to waitpid on {}: {:?}", child, e),
-                        }
-                    }
-                }
-                unistd::ForkResult::Child => {
-                    drop(exit_status_channel);
+        // Avoid clossing accidentally the channel file descriptors
+        self.file_descriptors(&[rt_tx.as_raw_fd(), rt_rx.as_raw_fd()]);
```

Comment:
The fd leak test is really our friend ;-)

Reply:
meaning? :sweat_smile: what's the problem here?